TDI-37924:Upgrade poi.jar to 3.16

pyzhou-talend opened the pull request:
https://jira.talendforge.org/browse/TDI-38105

pyzhou-talend commented:
I copy the link of TDI-38105 here on purpose, it's not a mistake.

pyzhou-talend commented:
I will definitely have another PR after the jar build by Jenkins. I replace the jar of talendMsgUtil only aim to not block the usage of tFileinputMail these days(as you know the devops issue may take weeks). the jar of talendMsgUtil will be eliminated on my next PR.

pyzhou-talend commented:
And thanks for the remaining, I will check it when new talend product build appear after integrate everything.

nrousseau commented on main/plugins/org.talend.designer.components.libs/libs_src/talendMsgMailUtil/pom.xml:
```diff
@@ -4,29 +4,90 @@
+                        </goals>
+                        <configuration>
+                            <outputDirectory>../../../org.talend.designer.components.localprovider/components/tFileInputMail</outputDirectory>
+                        </configuration>
```
That solution can't work at all, better change this... only could be usefull at worst from dev

nrousseau commented on main/plugins/org.talend.designer.components.localprovider/components/tFileInputMail/tFileInputMail_java.xml:
```diff
@@ -87,9 +87,9 @@
-      <IMPORT NAME="talendMsgMailUtil" MODULE="talendMsgMailUtil-1.0.0.jar" MVN="mvn:org.talend.libraries/talendMsgMailUtil-1.0.0/6.0.0"   REQUIRED_IF="(MAIL_TYPE == 'MSG')" />
+      <IMPORT NAME="poi" MODULE="poi-3.16-20170419_modified_talend.jar" MVN="mvn:org.talend.libraries/poi-3.16-20170419_modified_talend/6.0.0" UrlPath="platform:/plugin/org.talend.libraries.excel/lib/poi-3.16-20170419_modified_talend.jar" REQUIRED_IF="(MAIL_TYPE == 'MSG')" />
+      <IMPORT NAME="poi-scratchpad" MODULE="poi-scratchpad-3.16-20170419.jar" MVN="mvn:org.talend.libraries/poi-scratchpad-3.16-20170419/6.0.0" UrlPath="platform:/plugin/org.talend.libraries.excel/lib/poi-scratchpad-3.16-20170419.jar" REQUIRED_IF="(MAIL_TYPE == 'MSG')" />
+      <IMPORT NAME="talendMsgMailUtil" MODULE="talendMsgMailUtil-1.0-20170419.jar" MVN="mvn:org.talend.libraries/talendMsgMailUtil-1.0/6.0.0"   REQUIRED_IF="(MAIL_TYPE == 'MSG')" />
```
please have the same logic between jar name + mvn url, and take care about the versions.
old jar name was 1.0.0, means this one should be like 1.1.0 (if it breaks the compatibility at least).
Up to you for the date, but if we keep the date, then it should be in the final mvn url also

pyzhou-talend commented:
I have remove the binary of the mail jar, changed the version to 1.1 and removed the useless maven plugin in pom of talendMSGUtil.jar. I will change org.talend.designer.components.localprovider/pom.xml after DevOps issue done, cause I concern about the absent of talendMSGUtil.jar may lead to build fail. (the jar will be deployed on Nexus  automatic by Devops)
